src/init/features/dataconnect/sdk.ts

```diff
@@ -75,53 +102,52 @@ async function askQuestions(setup: Setup, config: Config): Promise<SDKInfo> {
-      type: "input",
-      default:
-        newConnectorYaml.generate.swiftSdk?.outputDir ||
-        `./../gensdk/${newConnectorYaml.connectorId}/swift-sdk`,
-    });
+  if (targetPlatform === Platform.IOS) {
+    const outputDir =
+      newConnectorYaml.generate.swiftSdk?.outputDir ||
+      path.relative(
+        connectorInfo.directory,
+        path.join(appDir, `generated/${newConnectorYaml.connectorId}`),
+      );
     const pkg = camelCase(newConnectorYaml.connectorId);
     const swiftSdk = { outputDir, package: pkg };
     newConnectorYaml.generate.swiftSdk = swiftSdk;
   }
-  if (platforms.includes(WEB)) {
-    const outputDir = await promptOnce({
-      message: `What directory do you want to write your JavaScript SDK code to? (If not absolute, path will be relative to '${connectorInfo.directory}')`,
-      type: "input",
-      default:
-        newConnectorYaml.generate.javascriptSdk?.outputDir ||
-        `./../gensdk/${newConnectorYaml.connectorId}/javascript-sdk`,
-    });
+
+  if (targetPlatform === Platform.WEB) {
+    const outputDir =
+      newConnectorYaml.generate.javascriptSdk?.outputDir ||
+      path.relative(
+        connectorInfo.directory,
+        path.join(appDir, `generated/${newConnectorYaml.connectorId}`),
+      );
     const pkg =
       newConnectorYaml.generate.javascriptSdk?.package ??
       `@firebasegen/${connectorInfo.connectorYaml.connectorId}`;
-    const packageJSONDir = await promptOnce({
-      message:
-        "Which directory contains the package.json that you would like to add the JavaScript SDK dependency to? (Leave blank to skip)",
-      type: "input",
-      default: newConnectorYaml.generate.javascriptSdk?.packageJSONDir,
-    });
-    // ../.. since we ask relative to connector.yaml
+
     const javascriptSdk: JavascriptSDK = {
       outputDir,
       package: pkg,
     };
-    if (packageJSONDir) {
-      javascriptSdk.packageJSONDir = packageJSONDir;
+
+    if (
+      (await directoryHasPackageJson(appDir)) &&
+      (await confirm({
+        message: "Would you like to add a dependency on the generated SDK to your package.json?",
+      }))
+    ) {
+      javascriptSdk.packageJsonDir = path.relative(connectorInfo.directory, appDir);
     }
     newConnectorYaml.generate.javascriptSdk = javascriptSdk;
   }
-  if (platforms.includes(ANDROID)) {
-    const outputDir = await promptOnce({
-      message: `What directory do you want to write your Kotlin SDK code to? (If not absolute, path will be relative to '${connectorInfo.directory}')`,
-      type: "input",
-      default:
-        newConnectorYaml.generate.kotlinSdk?.outputDir ||
-        `./../gensdk/${newConnectorYaml.connectorId}/kotlin-sdk`,
-    });
+
+  if (targetPlatform === Platform.ANDROID) {
+    const outputDir =
+      newConnectorYaml.generate.kotlinSdk?.outputDir ||
+      path.relative(
+        connectorInfo.directory,
+        path.join(appDir, `generated/${newConnectorYaml.connectorId}`),
```

Comment:
This is a good start, but for the vast majority of Android projects this output directory will be nonsensical. For small, single-module Android projects that use the conventional directory structure, a better output directory would be `app/src/main`. However, `app` is only a convention, and it can be named anything. Maybe consider looking for `app/src/main` and using that directory if it exists.

Reply:
Good idea - added a check for that directory. If its not found, should fall back to the appDir (the current behavior), or are there other structure we ought to try?